### ISSUE TITLE
Normalization not working as expected

### ISSUE THREAD

**Issue by perry-mitchell** [jonschlinkert/map-schema#4]
Considering the following example:

```javascript
const Schema = require("map-schema");

const TestSchema = new Schema();
TestSchema.field("bool", "boolean");

console.log(TestSchema.normalize({
  bool: 1
}));
```

I would expect that the `bool: 1` is changed to `bool: true`. But it stays as `1`. Am I missing something? I don't want to have to specify normalization functions for all records as that makes the _type_ parameter quite meaningless.

Hope I'm just missing something here :)

**Comment by doowb** [comment]
The `type` parameter is there for type checking, but it doesn't do automatic normalization.
However, it does cause a warning to be added to the `.warnings` array:

```js
console.log(TestSchema.warnings);
/*
[
  {
    method: 'invalidType',
    prop: 'bool',
    message: { expected: [ 'boolean' ], actual: 1 },
    args: undefined
  }
]
*/
```

I believe that explicitly defining normalizers is better in this situation because it allows you (the implementor) to choose how you handle types like that. For instance, as an implementer, I might choose to use the [affirmative](https://github.com/jonschlinkert/affirmative) and [falsey](https://github.com/jonschlinkert/falsey) packages to normalize to do more than just `Boolean(val)`:

```js
const falsey = require('falsey');
const affirmative = require('affirmative');

const Schema = require('map-schema');

const normalizeBoolean = val => {
  if (typeof val === 'boolean') return val;
  if (affirmative.includes(val)) return true;
  if (falsey(val)) return false;
  return false;
};

const TestSchema = new Schema()
  .field('bool_0', 'boolean', normalizeBoolean)
  .field('bool_1', 'boolean', normalizeBoolean)
  .field('bool_2', 'boolean', normalizeBoolean)
  .field('bool_true', 'boolean', normalizeBoolean)
  .field('bool_false', 'boolean', normalizeBoolean)
  .field('bool_yes', 'boolean', normalizeBoolean)
  .field('bool_no', 'boolean', normalizeBoolean);

const config = TestSchema.normalize({
  bool_0: 0,
  bool_1: 1,
  bool_2: 2,
  bool_true: true,
  bool_false: false,
  bool_yes: 'yes',
  bool_no: 'no'
});

console.log(config);
console.log(TestSchema.warnings);
/*
{
  bool_0: false,
  bool_1: false,
  bool_2: false,
  bool_true: true,
  bool_false: false,
  bool_yes: true,
  bool_no: false
}
*/
```

In this example, if a default normalizer used that looked like `val => Boolean(val)`, then the results would look like this: 

```js
{
  bool_0: false,
  bool_1: true,
  bool_2: true,
  bool_true: true,
  bool_false: false,
  bool_yes: true,
  bool_no: true
}
```

I think this leads to more ambiguity than being explicit about what you (as an implementer) thinks a boolean value should be.

BTW... I'm glad you opened this issue because it made me think about this a little more after my initial instinct of thinking, "of course `1` is `true`", because I wouldn't expect `2` to be `true` in most situations (like if the property was called `enabled`, what does `enabled: 2` mean?). Also `Boolean(0) === false` and `Boolean('0') === true`, so this could lead to other issues.
